Source/CombatExtended/Harmony/Harmony_ThingDef.cs

```diff
@@ -23,17 +23,16 @@ internal static class Harmony_ThingDef
         private const string CoverStatName = "CoverEffectiveness";
 
         private static System.Type type;
-        private static FieldInfo weaponField;
-        private static FieldInfo thisField;
-        private static FieldInfo currentField;
-        private static FieldInfo valueStringInt = AccessTools.Field(typeof(StatDrawEntry), "valueStringInt");
+        private static AccessTools.FieldRef<object, VerbProperties> weaponField;
+        private static AccessTools.FieldRef<object, ThingDef> thisField;
+        private static AccessTools.FieldRef<object, StatDrawEntry> currentField;
 
         static MethodBase TargetMethod()
         {
             type = typeof(ThingDef).GetNestedTypes(AccessTools.all).FirstOrDefault(x => x.Name.Contains("<SpecialDisplayStats>"));
-            weaponField = AccessTools.Field(type, AccessTools.GetFieldNames(type).FirstOrDefault(x => x.Contains("<verb>")));
-            thisField = AccessTools.Field(type, AccessTools.GetFieldNames(type).FirstOrDefault(x => x.Contains("this")));
-            currentField = AccessTools.Field(type, AccessTools.GetFieldNames(type).FirstOrDefault(x => x.Contains("current")));
+            weaponField = AccessTools.FieldRefAccess<VerbProperties>(type, AccessTools.GetFieldNames(type).FirstOrDefault(x => x.Contains("<verb>")));
```

Comment:
What is the advantage of `AccessTools.FieldRefAccess` rather than `AccessTools.Field`?  I see it is a template that takes the Type as a template parameter, but does it affect performance beyond that?

Reply:
The main benefit is performance. From my personal (and probably a bit too simple) tests, it ended up being around 10-20 times faster than working with `FieldInfo` directly.

The additional benefits are the removed need to cast and slightly more convenient to use syntax.

Reply:
The downside is the setup cost, as it needs to set it up. However, once it's set up it'll perform much better when operating on the fields.

Reply:
That makes sense.  Given the setup only runs once, the higher first time cost is not a significan issue.  Does mean if the code will be executed more than once, caching the `FieldRefaccess` will be critical, but that was the case with `FieldInfo` too.